lib/libzpool/uzfs_rebuilding.c

```diff
@@ -366,17 +413,36 @@ uzfs_zvol_create_snaprebuild_clone(zvol_state_t *zv,
 	    strchr(zv->zv_name, '/') + 1,
 	    REBUILD_SNAPSHOT_CLONENAME);
 
+	clone_subname = kmem_asprintf("%s_%s", strchr(zv->zv_name, '/') + 1,
+	    REBUILD_SNAPSHOT_CLONENAME);
+
 	ret = dmu_objset_clone(clonename, snapname);
-	if (ret == EEXIST) {
-		LOG_INFO("Volume:%s already has clone for snap rebuild",
-		    zv->zv_name);
+	if ((ret == EEXIST) || (ret == 0)) {
```

Comment:
its good that we left to zfs code to check for 'EEXIST' case by calling dmu_objset_clone first